### ISSUE TITLE
Fix/remove Docker tests

### ISSUE THREAD

**Issue by sohaibiftikhar** [FRosner/cluster-broccoli#369]
The Docker tests currently require docker hub credentials and any contribution made from a fork fails the test making the CI not very reliable unless you look at the logs. Maybe we should push to docker hub only when a request is merged to master.

**Comment by FRosner** [comment]
Yes the Docker push should only be executed if the branch is NOT a pull request coming from external. I think Travis injects an environment variable to determine whether it's an internal branch or an external PR.

We cannot push only on master as we make use of the tags from branches for test deployments as well.

Would you like to make that change, @sohaibiftikhar?
